src/Components/NetworkTable/NetworkTableBody.jsx

```diff
@@ -0,0 +1,77 @@
+import React, { useEffect, useRef } from 'react';
+import { FixedSizeList } from 'react-window';
+import PropTypes from 'prop-types';
+
+import { useNetwork } from '../../state/network/Context';
+import NetworkTableRow from './NetworkTableRow';
+import { TABLE_ENTRY_HEIGHT } from '../../constants';
+import { useResizeObserver } from '../../hooks/useResizeObserver';
+
+const virtualizedTableRow = ({
+  data,
+  index,
+  style,
+}) => {
+  const {
+    listData,
+    totalNetworkTime,
+    handleReqSelect,
+    selectedReqIndex,
+  } = data;
+  const item = listData.get(index);
+
+  return (
+    <NetworkTableRow
+      key={index}
+      entry={item}
+      maxTime={totalNetworkTime}
+      onSelect={handleReqSelect}
+      scrollHighlight={selectedReqIndex === index}
+      style={style}
+    />
+  );
+};
+
+const NetworkTableBody = ({ height }) => {
+  const {
+    state,
+    actions,
+    callbacks,
+  } = useNetwork();
+  const data = state.get('data');
+  const totalNetworkTime = state.get('totalNetworkTime');
+  const selectedReqIndex = state.get('selectedReqIndex');
+
+  const ref = useRef(null);
+  const { elementDims } = useResizeObserver(ref);
```

Comment:
You can use [Autosizer](https://github.com/bvaughn/react-window?tab=readme-ov-file#related-libraries) from `react-window` that is made to handle dynamic width and height,

Reply:
I tried using Autosizer for the virtualized table dimensions but it didn't work so well with the current layout. 

The ResizeObserver here is used to get the width of the table-body to set it for the table-header. This is necessary because of the scrollbar that can appear for the table-body causing the width of the body to the smaller than the header.